frontend/src/components/custom-tools/export-tool/ExportTool.jsx

```diff
@@ -0,0 +1,228 @@
+import {
+  DeleteOutlined,
+  QuestionCircleOutlined,
+  UserOutlined,
+} from "@ant-design/icons";
+import {
+  Avatar,
+  List,
+  Modal,
+  Popconfirm,
+  Radio,
+  Select,
+  Typography,
+} from "antd";
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import "./ExportTool.css";
+
+import { SpinnerLoader } from "../../widgets/spinner-loader/SpinnerLoader";
+
+const SHARE_ALL = "share_all";
+const SHARE_CUSTOM = "share_custom";
+
+function ExportTool({
+  open,
+  setOpen,
+  toolDetails,
+
+  loading,
+  allUsers,
+  onApply,
+}) {
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [filteredUserList, setFilteredUserList] = useState([]);
+  const [sharingOption, setSharingOption] = useState(SHARE_ALL);
+
+  useEffect(() => {
+    if (toolDetails?.shared_users) {
+      // set the selectedUsers to the IDs of shared users
+      // filter owners and shared users
+
+      const promptStudioUsers = toolDetails?.prompt_studio_users.map((user) =>
+        user?.id?.toString()
+      );
+
+      const users = allUsers.filter((user) => {
+        const userId = user?.id?.toString();
+        return (
+          !selectedUsers.includes(userId) &&
+          !promptStudioUsers.includes(userId) &&
+          toolDetails?.created_by?.toString() !== userId
+        );
+      });
+
+      setFilteredUserList(users);
+      toolDetails?.shared_to_org
+        ? setSharingOption(SHARE_ALL)
+        : setSharingOption(SHARE_CUSTOM);
+    }
+  }, [toolDetails, allUsers, selectedUsers]);
```

Comment:
@johnyrahul We already have another `useEffect` (line 62) with dependencies of `toolDetails` and `allUsers`, where we run some logic and set the `selectedUsers`. Do we need the same dependencies here as well? If yes, then can we try and merge these two `useEffects` to enhance efficiency?

Reply:
@jagadeeswaran-zipstack  will require your help here.
Copied the similar logic from the share permissions component. Can you help to optimize this?

Reply:
Sure. I'll make the required changes